kratos/sources/model_part_io.cpp

```diff
@@ -1775,6 +1776,67 @@ namespace Kratos
         KRATOS_CATCH("")
     }
 
+    void ModelPartIO::WriteElementalDataBlock(ElementsContainerType& rThisElements){
+        typedef VariableComponent<VectorComponentAdaptor<array_1d<double, 3> > > array_1d_component_type;
+        std::unordered_set<std::string> variables;
+        
+        for(auto& elem :rThisElements){
+            for(auto& var:elem.GetData()){
+                auto is_included = variables.find(var.first->Name());
+                if(is_included == variables.end()){
+                    variables.insert(var.first->Name());
+                    // determine variable type
+                    if(KratosComponents<Variable<bool>>::Has(var.first->Name())){
+                        Variable<bool> variable = KratosComponents<Variable<bool>>::Get(var.first->Name());
+                        WriteElementalDataBlock<Variable<bool>>(rThisElements, variable);
+                    }
+                    else if(KratosComponents<Variable<int>>::Has(var.first->Name())){
```

Comment:
I would move the body of all these else ifs to a separate (private) template method `WriteElementalDataIfTypeIs< >` to avoid repeatition

Reply:
@annasophie91 this is also what I would have suggested 👍

Reply:
I integrated it in the already existing function WriteDataBlock<>() to avoid another function